Adventurer.cpp

```diff
@@ -16,7 +16,9 @@ void Adventurer::increaseXP(int xpAmmount)
 	}
 }
 
-void Adventurer::attackEnemy(Character& opponent){
-	int damageDealt = opponent.takeDamage(*this);
+int Adventurer::attackEnemy(Character& opponent){
+	int damageDealt = Character::attackEnemy(opponent);
+	this->currentCooldown += this->cooldown;

```

Comment:
Ez benne van a `Character::attackEnemy`ben is. Így duplán adódik hozzá a cooldown, ha jól látom.